src/MatrixEventHandler.ts

```diff
@@ -250,46 +251,74 @@ export class MatrixEventHandler {
         }
     }
 
+    public async onTyping(r: Request<TypingEvent>) {
+        const typing = r.getData();
+        const ctx: RoomBridgeStoreEntry = (await this.store.getRoomEntryByMatrixId(typing.room_id) || {
+            matrix: undefined,
+            remote: undefined,
+            data: {},
+        });
+        const roomType: string|null = ctx.matrix ? ctx.matrix.get("type") : null;
+
+        if (roomType === MROOM_TYPE_IM) {
+            // We only support this on IM rooms for now.
+            // Assuming only one Matrix user present.
+            log.info(`Handling IM typing for ${typing.room_id}`);
+            if (!ctx.remote) {
+                throw Error('Cannot handle message, remote or matrix not defined');
+            }
+            const isUserTyping = !!typing.content.user_ids.filter(u => !this.bridge.getBot().isRemoteUser(u))[0];
+            const matrixUser = ctx.remote.get<string>("matrixUser");
+            let acct: IBifrostAccount;
+            const roomProtocol: string = ctx.remote.get("protocol_id");
+            try {
+                acct = (await this.getAccountForMxid(matrixUser, roomProtocol)).acct;
+            } catch (ex) {
+                log.error(`Couldn't handle ${matrixUser}'s typing event, ${ex}`);
+                return;
+            }
+            const recipient: string = ctx.remote.get("recipient");
+            log.info(`Sending typing to ${recipient}`);
+            acct.sendIMTyping(recipient, isUserTyping);
+            return;
+        }
+    }
+
     /* NOTE: Command handling should really be it's own class, but I am cutting corners.*/
     private async handleCommand(args: string[], event: WeakEvent) {
         if (!this.bridge) {
             throw Error('Bridge is not defined');
         }
+        // Are we running a bridge for one protocol?
+        const soloProtocol = this.purple.usingSingleProtocol();
```

Comment:
We badly need to refactor the admin room code below but anyway: This change tries to simplify the command interface by removing functionality that would only be needed for multi-protocol support.